SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LayeredLBStrategy.cpp

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/LayeredLBStrategy.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include "Utils/LayerInfo.h"
+#include "Utils/SpatialActorUtils.h"
+
+#include "Templates/Tuple.h"
+
+DEFINE_LOG_CATEGORY(LogLayeredLBStrategy);
+
+ULayeredLBStrategy::ULayeredLBStrategy()
+	: Super()
+{
+}
+
+ULayeredLBStrategy::~ULayeredLBStrategy()
+{
+	for (const auto& Elem : LayerNameToLBStrategy)
+	{
+		Elem.Value->RemoveFromRoot();
+	}
+}
+
+void ULayeredLBStrategy::Init()
+{
+	Super::Init();
+
+	VirtualWorkerId CurrentVirtualWorkerId = SpatialConstants::INVALID_VIRTUAL_WORKER_ID + 1;
+
+	if (const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>())
+	{
+		check(Settings->bEnableUnrealLoadBalancer);
+		const ASpatialWorldSettings* WorldSettings = GetWorld() ? Cast<ASpatialWorldSettings>(GetWorld()->GetWorldSettings()) : nullptr;
+
+		if (WorldSettings == nullptr)
+		{
+			UE_LOG(LogLayeredLBStrategy, Error, TEXT("If EnableUnrealLoadBalancer is set, WorldSettings should inherit from SpatialWorldSettings to get the load balancing strategy."));
+			UAbstractLBStrategy* DefaultLBStrategy = NewObject<UGridBasedLBStrategy>(this);
+			DefaultLBStrategy->AddToRoot();
```

Comment:
Might be nifty to wrap up adding adding LBS to LayerNameToLBStrategy: AddLBStrategyLayor(FName LayorName,<Sublass AstactLBS> Class)

It could: 
call AddToRoot()
call Init()
and add the LBS to LayerNameToLBStrategy

Reply:
Ooooh yeah this gonna be nice.

Reply:
Good idea, I hadn't realized how many times I was doing that.